inc/vcf/error.hpp

```diff
@@ -256,32 +260,48 @@ namespace ebi
     #pragma db object
     struct FilterBodyError : public BodySectionError
     {
-        using BodySectionError::BodySectionError;
-        FilterBodyError() : FilterBodyError{0} {}
-        FilterBodyError(size_t line) : FilterBodyError{line, "Filter is not a single dot or a semicolon-separated list of strings"} { }
+        FilterBodyError(size_t line = 0,
+                        const std::string &message = "Filter is not a single dot or a semicolon-separated list of strings",
+                        ErrorFix error_fix = ErrorFix::IRRECOVERABLE_VALUE,
+                        const std::string &field = "")
+                : BodySectionError{line, message}, error_fix{error_fix}, field{field} { }
         virtual ~FilterBodyError() override { }
         virtual void apply_visitor(ErrorVisitor &visitor) override { visitor.visit(*this); }
+
+        ErrorFix error_fix;
+        const std::string field;
     };
     #pragma db object
     struct InfoBodyError : public BodySectionError
     {
         InfoBodyError(size_t line = 0,
                       const std::string &message = "Error in info column, in body section",
-                      const std::string &field = "")
-                : BodySectionError{line, message}, field{field} {}
+                      ErrorFix error_fix = ErrorFix::IRRECOVERABLE_VALUE,
+                      const std::string &field = "",
+                      const std::string &expected_value = "")
+                : BodySectionError{line, message}, error_fix{error_fix}, field{field}, expected_value{expected_value} {
+                    if (error_fix == ErrorFix::RECOVERABLE_VALUE && expected_value.empty()) {
```

Comment:
This is more a std::logic_error that a invalid argument. if this situation occurs, it is likely to be due to a buggy code. You can also extract this check into a function, as it should be also in FilterBodyError and IdBodyError. I think the prototype could be:
```c++
void assert_recoverable_value_is_provided(ErrorFix error_fix, const std::string &expected_value);
```
And the message could start with "The program had an internal error: "